### PR TITLE
OSDOCS-1142: Adding priority and namespace filtering

### PR DESCRIPTION
**Pull request by bergerhoffer**
https://issues.redhat.com/browse/OSDOCS-1142

Preview: 
* https://osdocs-1142-filtering--ocpdocs.netlify.app/openshift-enterprise/latest/nodes/scheduling/nodes-descheduler.html#nodes-descheduler-filtering-namespace_nodes-descheduler
* https://osdocs-1142-filtering--ocpdocs.netlify.app/openshift-enterprise/latest/nodes/scheduling/nodes-descheduler.html#nodes-descheduler-filtering-priority_nodes-descheduler

### CODE REVIEW

**Review by damemi — APPROVED** [review]
/lgtm
from the descheduler side

**Review by sfortner-RH — APPROVED** [review]
LGTM

File: modules/nodes-descheduler-filtering-priority.adoc [hunk 1]
```diff
@@ -0,0 +1,54 @@
+// Module included in the following assemblies:
+//
+// * nodes/scheduling/nodes-descheduler.adoc
+
+[id="nodes-descheduler-filtering-priority_{context}"]
+= Filtering pods by priority
+
+You can configure descheduler strategies to only consider pods for eviction if their priority is lower than a specified priority level. Pods that are higher than the specified priority threshold are not considered for eviction.
```
Comment (kalexand-rh) [inline modules/nodes-descheduler-filtering-priority.adoc:8]:
```suggestion
You can configure descheduler strategies to consider pods for eviction only if their priority is lower than a specified priority level. Pods that are higher than the specified priority threshold are not considered for eviction.
```

File: modules/nodes-descheduler-filtering-namespace.adoc [hunk 2]
```diff
@@ -0,0 +1,61 @@
+// Module included in the following assemblies:
+//
+// * nodes/scheduling/nodes-descheduler.adoc
+
+[id="nodes-descheduler-filtering-namespace_{context}"]
+= Filtering pods by namespace
+
+You can configure whether or not pods should be considered for eviction based on their namespace. Only the following descheduler strategies support namespace filtering:
```
Comment (kalexand-rh) [inline modules/nodes-descheduler-filtering-namespace.adoc:8]:
```suggestion
You can configure whether or not pods are considered for eviction based on their namespace. Only the following descheduler strategies support namespace filtering:
```

**Review by kalexand-rh — COMMENTED** [review]
Two small things, and then this LGTM.

File: modules/nodes-descheduler-filtering-namespace.adoc [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+You can configure whether or not pods are considered for eviction based on their namespace. Only the following descheduler strategies support namespace filtering:
+
+* `PodLifeTime`
+* `RemoveDuplicates`
+* `RemovePodsHavingTooManyRestarts`
+* `RemovePodsViolatingInterPodAntiAffinity`
+* `RemovePodsViolatingNodeAffinity`
+* `RemovePodsViolatingNodeTaints`
+
```
Comment (kasturinarra) [inline modules/nodes-descheduler-filtering-namespace.adoc:16]: @bergerhoffer  can we remove `RemoveDuplicates` and add `RemovePodsHavingTooManyRestarts`
RemoveDuplicates and LowNodeUtilization does not support NamespaceFiltering.

File: modules/nodes-descheduler-filtering-priority.adoc [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    params:
+    - name: "MaxPodLifeTimeSeconds"
+      value: "86400"
+    - name: "ThresholdPriorityClassName" <1>
+      value: "my-priority-class-name" <2>
+----
+<1> You cannot specify both `ThresholdPriority` and `ThresholdPriorityClassName` for the same strategy.
+<2> The numerical priority value associated with this priority class name is used as the threshold.
+
+. Save the file to apply the changes.
```
Comment (kasturinarra) [inline modules/nodes-descheduler-filtering-priority.adoc:54]: May be it is good to add the below point here ?
 if the given priority class does not exist, descheduler won't create it and will throw an error.

File: modules/nodes-descheduler-filtering-namespace.adoc (hunk 3 above)
Comment (bergerhoffer) [reply modules/nodes-descheduler-filtering-namespace.adoc:16]: @kasturinarra I got this list from here, and it lists `RemoveDuplicates`: https://github.com/kubernetes-sigs/descheduler#namespace-filtering

Does OpenShift's descheduler differ? CC @damemi 

Also `RemovePodsHavingTooManyRestarts` is already in the list.

Comment (bergerhoffer) [reply modules/nodes-descheduler-filtering-namespace.adoc:15]: @damemi Confirmed with me that `RemoveDuplicates` was only recently added, and so not a part of 4.6. Removing!

Comment (kasturinarra) [reply modules/nodes-descheduler-filtering-namespace.adoc:15]: Thanks !!

File: modules/nodes-descheduler-filtering-priority.adoc (hunk 4 above)
Comment (kasturinarra) [reply modules/nodes-descheduler-filtering-priority.adoc:54]: Thanks !!

**Comment by kasturinarra** [comment]
Reviewed again and i could see the changes. Thanks !!

**Pull request merged**